### ISSUE TITLE
Particles per second parameter apparently does not change anything

### ISSUE THREAD

**Issue by ANPez** [DanielMartinus/Konfetti#20]
I see the same result when passing "1" as parameters as when 1000.

**Comment by DanielMartinus** [comment]
Thanks for reporting @ANPez 

I did a quick look in the code and as far as I can see is that `amountPerMs` is calculated when the StreamEmitter is created here:
https://github.com/DanielMartinus/Konfetti/blob/c242aecac9e712cf753d102f64ab0fb3d7d58f2d/konfetti/src/main/java/nl/dionsegijn/konfetti/emitters/StreamEmitter.kt#L29

But not used when the particles are being created here
https://github.com/DanielMartinus/Konfetti/blob/c242aecac9e712cf753d102f64ab0fb3d7d58f2d/konfetti/src/main/java/nl/dionsegijn/konfetti/emitters/StreamEmitter.kt#L33-L51

I'll make some time to implement this properly and see if I can cover this with tests. But I have not much time until the weekend. A Pull request is welcome ofcourse if there is no time to wait.

In the meantime you might be able to achieve the same by creating more than one particle stream.

**Comment by DanielMartinus** [comment]
Hi @ANPez,

To fix the particles per second bug I've refactored the `StreamEmitter`. I've opened a pull request which also contains a couple of other fixes such as rewriting the RenderSystem to make it all more testable. You'll also find tests in the pull request referenced in this issue.

I'm not exactly sure when I'll push an update for the library, for now the pull request isn't merged yet. I'll update you on the progress.
